### DISCUSSION TITLE
bookstore performance demo - really small factors and even reverse results...

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#592] [General]
Hi all;
I have deployed the bookstore performance demo on my local machine. it a good laptop (11th gen i7, 4 core, 2.8Gh with 32G RAM)
given that there are other things running on the laptop that affect both the microstream scenario and the postgres one (even playing field), many of the results give a factor of 1-10 only in favor ot microstream. some of the results, specifically for the 'Employee of the year', small factor results in favor of JPA for some iterations (consistently). 
is this something to expect (low factor and sometimes losing)? 
this is not alligned to the stats show on https://web.bookstore.demo.microstream.one/ 
your comments are welcomed...

**Comment by fh-ms** [comment]
Hi, what data amount did you use? The demo on the website is based on the largest one.
https://github.com/microstream-one/bookstore-demo-performance/blob/master/src/main/resources/application.properties#L10

**Reply by guykatz** [reply]
hi @fh-ms 
I used the defaults (medium)... the only thing I changed in the connection string and thats it.
should I edit the data size and recheck? I would expect medium to have a big impact also...

**Reply by fh-ms** [reply]
It depends heavily on the used hardware and data model. The bigger and more complex the data gets, the bigger the impact will be.

**Reply by guykatz** [reply]
ill test with large data... thanks

**Reply by guykatz** [reply]
ok large made a difference in some queries but not all...
@fh-ms is the JPA variant has a 2nd level cache? ehcache or something similar?
thanks

**Reply by fh-ms** [reply]
Yes, caching is enabled:
https://github.com/microstream-one/bookstore-demo-performance/blob/master/src/main/resources/application.properties#L37C3-L37C68
https://github.com/microstream-one/bookstore-demo-performance/blob/master/pom.xml#L120

**Reply by guykatz** [reply]
@fh-ms is the live demo on https://web.bookstore.demo.microstream.one/ doing real online processing and showing the results?
